src/com/amazon/ion/impl/lite/IonSystemLite.java

```diff
@@ -822,4 +685,190 @@ public boolean valueIsSharedSymbolTable(IonValue value)
+                    return newBlob(reader.newBytes());
+                case LIST:
+                    return newEmptyList();
+                case SEXP:
+                    return newEmptySexp();
+                case STRUCT:
+                    return newEmptyStruct();
+                default:
+                    // Includes the variants `NULL` (handled prior to the switch) and `DATAGRAM`.
+                    throw new IonException("unexpected type encountered reading value: " + ionType);
+            }
+        }
+
+        // If the reader is positioned inside a struct, copies the field name to `value`.
+        // Note that this will NOT copy the field name over if the reader was inside a struct when `load(reader)` was
+        // called. For example, if a reader is consuming the following data:
+        //
+        //   {
+        //     foo: [1, 2, 3]
+        //   }
+        //
+        // And the reader is positioned on the field value `[1, 2, 3]` when the `load(reader)` was called, this
+        // method will NOT copy the field name `foo` over. ValueLoader treats the reader's initial state as the
+        // "top level" for the purposes of materializing the current value.
+        // For a test that enforces this behavior, see: IonReaderToIonValueTest.
+        // Returns `true` if the reader's current value has a field name; otherwise, returns false.
+        private boolean cloneFieldNameIfAny(IonValueLite value) {
+            if (containerStack.isEmpty() || !reader.isInStruct()) {
+                // This value is in a context that doesn't have a field name.
+                return false;
+            }
+            SymbolToken token = reader.getFieldNameSymbol();
+            String text = token.getText();
+            if (text != null && token.getSid() != UNKNOWN_SYMBOL_ID)
+            {
+                token = newSymbolToken(text, UNKNOWN_SYMBOL_ID);
+            }
+            value.setFieldNameSymbol(token);
+            return true;
+        }
+
+        // Copies any annotations found on the reader's current position over to the provided `value`.
+        // Returns `true` if more than any annotations were found/copied. If no annotations were found/copied,
+        // returns `false`.
+        private boolean cloneAnnotationsIfAny(IonValueLite value) {
+            // `getTypeAnnotationSymbols` returns a freshly allocated array, so we can safely modify it.
+            SymbolToken[] annotations = reader.getTypeAnnotationSymbols();
+            if (annotations.length == 0) {
+                return false;
+            }
+
+            for (int i = 0; i < annotations.length; i++)
+            {
+                SymbolToken token = annotations[i];
+                String text = token.getText();
+                if (text != null && token.getSid() != UNKNOWN_SYMBOL_ID )
+                {
+                    annotations[i] = newSymbolToken(text, UNKNOWN_SYMBOL_ID);
+                }
+            }
```

Comment:
I know you didn't change this, but I'm noticing that it's fairly expensive, and doesn't provide any benefit as far as I can see. It removes the local SID for symbols whose text is defined, but any local SID should be ignored when text is defined anyway. The same local SID deletion is performed for field names, too.

Reply:
That's a good observation; I think this is intended to be a defensive copy, but `SymbolToken` is immutable so we don't actually need one. A shallow clone will do. I've opened #526 to track this.